modules/integration/src/test/scala/scala/cli/integration/SparkTests212.scala

```diff
@@ -8,34 +8,69 @@ import scala.util.Properties
 
 object SparkTests212 {
 
-  private final case class Spark(sparkVersion: String, scalaVersion: String) {
-    def sparkHome(): os.Path = {
-      val url =
-        s"https://archive.apache.org/dist/spark/spark-$sparkVersion/spark-$sparkVersion-bin-hadoop2.7.tgz"
-      val dirName = url.drop(url.lastIndexOf('/') + 1).stripSuffix(".tgz")
+  private def lightweightSparkDistribVersionOpt = Option("0.0.3")
+
+  private final class Spark(val sparkVersion: String, val scalaVersion: String) {
+    private def sbv         = scalaVersion.split('.').take(2).mkString(".")
+    private var toDeleteOpt = Option.empty[os.Path]
+    lazy val sparkHome: os.Path = {
+      val url = lightweightSparkDistribVersionOpt match {
+        case Some(lightweightSparkDistribVersion) =>
+          s"https://github.com/scala-cli/lightweight-spark-distrib/releases/download/v$lightweightSparkDistribVersion/spark-$sparkVersion-bin-hadoop2.7-scala$sbv.tgz"
+        case None =>
+          // original URL (too heavyweight, often fails / times out…)
+          s"https://archive.apache.org/dist/spark/spark-$sparkVersion/spark-$sparkVersion-bin-hadoop2.7.tgz"
+      }
       val baseDir =
         os.Path(os.proc(TestUtil.cs, "get", "--archive", url).call().out.text().trim, os.pwd)
-      baseDir / dirName
+      val home = os.list(baseDir) match {
+        case Seq(dir) if os.isDir(dir) => dir
+        case _                         => baseDir
+      }
+      if (lightweightSparkDistribVersionOpt.nonEmpty) {
+        val copy = os.temp.dir(prefix = home.last) / "home"
+        toDeleteOpt = Some(copy)
+        System.err.println(s"Copying $home over to $copy")
+        os.copy(home, copy)
+        val fetchJarsScript0 = copy / "fetch-jars.sh"
+        val cmd: Seq[os.Shellable] =
+          if (Properties.isWin) Seq("bash.exe", fetchJarsScript0)
+          else Seq(fetchJarsScript0)
+
+        System.err.println(s"Running $cmd")
+        os.proc(cmd).call(stdin = os.Inherit, stdout = os.Inherit)
+        System.err.println(s"Spark home $copy ready")
+        copy
+      }
+      else
+        home
     }
+    def cleanUp(): Unit =
+      toDeleteOpt.foreach(os.remove.all(_))
   }
 
-  private val spark30 = Spark(
+}
+
+class SparkTests212 extends SparkTestDefinitions {
+
+  import SparkTests212._
+
+  private val spark30 = new Spark(
     "3.0.3",
     // The spark distribution actually ships with Scala 2.12.10, but we run into #1092 if we use it here
     "2.12.15"
   )
 
-  private val spark24 = Spark(
+  private val spark24 = new Spark(
```

Comment:
NIT: the `new` keyword is actually redundant now, since we compile integration tests with Scala 3 now, no?

Reply:
Right. Fixed it, although I'm not fond of dropping `new` for non-case classes…

Reply:
Actually, I added back the `new`, it seems [they're still required](https://github.com/VirtusLab/scala-cli/runs/7539539013?check_suite_focus=true#step:7:957).